src/platform/cmx/kernel.c

```diff
@@ -0,0 +1,271 @@
+#include "assert.h"
+#include "kernel.h"
+#include "string.h"
+
+#include <captbl.h>
+#include <retype_tbl.h>
+#include <component.h>
+#include <thd.h>
+#include "stm32f7xx_hal.h"
+#include "core_cm7.h"
+#include "boot_comp.h"
+
+#include "user/cos_component.h"
+#include "mem_layout.h"
+
+#include "lcd.h"
+
+struct mem_layout glb_memlayout;
+volatile unsigned long long rdtsc_sim=1;
+
+/* We have 64 pages in the system now. Thus, the 512kB will be cut into 8kB pages. All the access should be aligned to 8kB */
+extern u32_t Stack_Mem[ALL_STACK_SZ];
+/* These unused memory should be assigned to the last of the space */
+extern unsigned long Memory_Used[65536];
```

Comment:
Again from above.  Can we hide these symbols in another file (perhaps `util.h`) so that if we go to another arm device with different backing libraries, we can make it work?  Generally how can we support multiple boards/manufacturers?